"Warning: control reaches end of non-void function." - Should I worry about that?

While compiling in Ubuntu 18.10 I get:

    /fc/src/log/log_message.cpp:165:4: warning: control reaches end of non-void function [-Wreturn-type]
    }

--- a/src/log/log_message.cpp
+++ b/src/log/log_message.cpp
@@ -162,9 +162,7 @@ namespace fc
          case log_level::off:
             return "off";
       }
-#ifdef __GNUC__
-      __builtin_unreachable();
-#endif
+      return "unknown";
    }
 
    string     log_context::get_file()const       { return my->file; }

